### ISSUE TITLE
create e2e test to ensure rerunning kfctl apply a 2nd time works kubeflow/kubeflow#3810

### ISSUE THREAD

**Issue by kkasravi** [kubeflow/kfctl#12]
issue kubeflow/kubeflow#3810 is

Running kfctl apply k8s fails on 2nd application #3810

**Comment by jlewi** [comment]
I'm bumping this to P0 for 0.7.

#3810 was a big issue for 0.6 and we should ensure we have an E2E test to verify this is fixed in 0.7.

**Comment by jlewi** [comment]
#36 is going to split kfctl_go_test.py into separate tests for building and deploying kubeflow. After that issue it should be relatively straightforward to add a step to our E2E workflow to rerun apply and verify that it doesn't error out.

**Comment by jlewi** [comment]
I manually verified that this was working with GCP IAP
kubeflow/manifests - v0.7.0-rc.2-16-gd6dad08
kubeflow/kubeflow - v0.4.0-rc.1-802-gd66f8b97

Rerunning kfctl apply a second time worked.

Immediately after running apply the 2nd time I started getting Origin Authentication Errors when accessing the centraldashboard.

It looks like reapplying the ingress causes the backend to change so the JWT policy needs to change.

The IAP enabler pod checks the JWT policy every 5 minutes so after 5 minutes the JWT policy is updated and the centraldashboard becomes available.

Manually deleting the pod would force the JWT policy to be updated immediately.

This JWT behavior is working as intended.

So I'm closing this as verified.
